aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -188,6 +192,27 @@ module aptos_framework::code {
         publish_package(owner, util::from_bytes<PackageMetadata>(metadata_serialized), code)
     }
 
+    /// Same as `publish_package_txn` but can dynamically replace addresses in the package with new ones.
+    /// The `address_mapping` must have even number of addresses, such as [A, B, C, D], and A and C in original package
+    /// would be replaced by B and D, respectively. If the same address is replaced more than once, the last mapping
+    /// would overwrite the previous ones.
+    public entry fun publish_package_with_address_remapping_txn(
+        owner: &signer,
+        metadata_serialized: vector<u8>,
+        code: vector<vector<u8>>,
+        address_mapping: vector<address>,
```

Comment:
This feels sort of error prone, can you do a vector of vectors instead? So each inner vector must have 2 elements, the current address and the address to remap to.